[bug] Conan build fails to find CMakeLists when run in the root level of a drive

### Describe the bug

### Environment Details
Operating System+version: Windows 10.0.19045
Compiler+version: cmake version 3.26.4
Conan version: 2.0.13
Python version: 3.11.6

### Description
We're running into problems related to path length for some builds on Windows. To mitigate this, we're making use of `subst` to shorten windows path lengths by remapping the path of the project directory to the root of a drive. If conan build is run in the root of this drive, it will fail to find CMakeLists, the same conan build will work in any other directory.

### How to reproduce it

### Commands
 1. conan new cmake_lib -d name=foo -d version=0.1.0
 2. subst x\: .
 3. cd x:
 4. conan build .
 5. Observe that conan cannot identify a CMakeLists.txt in this directory

### Logs
```
======== Input profiles ========
Profile host:
[settings]
arch=x86_64
build_type=Release
compiler=msvc
compiler.cppstd=17
compiler.runtime=dynamic
compiler.runtime_type=Release
compiler.version=193
os=Windows
[conf]
tools.cmake.cmaketoolchain:generator=NinjaProfile build:
[settings]
arch=x86_64
build_type=Release
compiler=msvc
compiler.cppstd=17
compiler.runtime=dynamic
compiler.runtime_type=Release
compiler.version=193
os=Windows
[conf]
tools.cmake.cmaketoolchain:generator=Ninja
======== Computing dependency graph ========
Graph root
    conanfile.py (foo/0.1.0): X:\conanfile.py
======== Computing necessary packages ========
======== Installing packages ========
======== Installing packages ========
======== Finalizing install (deploy, generators) ========
conanfile.py (foo/0.1.0): Calling generate()
conanfile.py (foo/0.1.0): Generators folder: X:\build\Release\generators
conanfile.py (foo/0.1.0): CMakeToolchain generated: conan_toolchain.cmake
conanfile.py (foo/0.1.0): Preset 'conan-release' added to CMakePresets.json. Invoke it manually using 'cmake --preset conan-release' if
 using CMake>=3.23
conanfile.py (foo/0.1.0): If your CMake version is not compatible with CMakePresets (<3.23) call cmake like: 'cmake <path> -G Ninja -DC
MAKE_TOOLCHAIN_FILE=X:\build\Release\generators\conan_toolchain.cmake -DCMAKE_POLICY_DEFAULT_CMP0091=NEW -DCMAKE_BUILD_TYPE=Release'
conanfile.py (foo/0.1.0): CMakeToolchain generated: CMakePresets.json
conanfile.py (foo/0.1.0): CMakeToolchain generated: ..\..\..\CMakeUserPresets.json
conanfile.py (foo/0.1.0): Generating aggregated env files
conanfile.py (foo/0.1.0): Generated aggregated env files: ['conanbuild.bat', 'conanrun.bat']
======== Calling build() ========
conanfile.py (foo/0.1.0): Calling build()
conanfile.py (foo/0.1.0): Running CMake.configure()
conanfile.py (foo/0.1.0): RUN: cmake -G "Ninja" -DCMAKE_TOOLCHAIN_FILE="X:/build/Release/generators/conan_toolchain.cmake" -DCMAKE_INST
ALL_PREFIX="X:/" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Release" "X:\"
conanvcvars.bat: Activating environment Visual Studio 17 - amd64 - vcvars_ver=14.3
[vcvarsall.bat] Environment initialized for: 'x64'
CMake Error: The source directory "X:." does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.
*********************************************************
Recipe 'conanfile.py (foo/0.1.0)' cannot build its binary
It is possible that this recipe is not Conan 2.0 ready
If the recipe comes from ConanCenter, report it at https://github.com/conan-io/conan-center-index/issues
If it is your recipe, check if it is updated to 2.0
*********************************************************
ERROR: conanfile.py (foo/0.1.0): Error in build() method, line 44
        cmake.configure()
        ConanException: Error 1 while executing
```

--- a/conan/tools/cmake/cmake.py
+++ b/conan/tools/cmake/cmake.py
@@ -85,10 +85,9 @@ class CMake(object):
         cmakelist_folder = self._conanfile.source_folder
         if build_script_folder:
             cmakelist_folder = os.path.join(self._conanfile.source_folder, build_script_folder)
+        cmakelist_folder = cmakelist_folder.replace("\\", "/")
 
         build_folder = self._conanfile.build_folder
-        generator_folder = self._conanfile.generators_folder
-
         mkdir(self._conanfile, build_folder)
 
         arg_list = [self._cmake_program]

--- a/conans/test/unittests/tools/cmake/test_cmake_presets_definitions.py
+++ b/conans/test/unittests/tools/cmake/test_cmake_presets_definitions.py
@@ -45,6 +45,7 @@ def test_cmake_make_program(conanfile):
         assert '-DCMAKE_MAKE_PROGRAM="C:/mymake.exe"' in command
 
     conanfile.run = run
+    conanfile.folders.set_base_source(temp_folder())
     conanfile.conf.define("tools.gnu:make_program", "C:\\mymake.exe")
 
     with mock.patch("platform.system", mock.MagicMock(return_value='Windows')):

